### PR TITLE
fix(telegram): pollingTimeout `socket hang up` error, fix #67

### LINKED ISSUE #67: bug(telegram): failed to get updates. reason: socket hang up
复现：
* 在 koishi 项目下运行 `yarn dev`
* 可见控制台每隔一段时间输出标题中的错误信息

我这里把 timeout 改成 25 就解决了，改成 30 也会报错。（原本是 60000）

```
const updates = await bot.internal.getUpdates({
            offset: this.offset + 1,
            timeout: 25
})
```

"**Timeout in seconds** for long polling. Defaults to 0, i.e. usual short polling. Should be positive, short polling should be used for testing purposes only."

node: 19.5.0
plugin: 3.6.5
koishi: 4.11.4

### PR DESCRIPTION
**Pull request by idanran**
Fixes #67

### CODE REVIEW

**Comment by shigma** [comment]
cc @Anillc Any comments for this change?

**Review by Anillc — APPROVED** [review]
looks ok

File: adapters/telegram/src/polling.ts [hunk 1]
```diff
@@ -83,7 +83,7 @@ export namespace HttpPolling {
 
   export const Config: Schema<Config> = Schema.object({
     protocol: Schema.const('polling' as const).required(),
-    pollingTimeout: Schema.natural().role('time').default(Time.minute).description('通过长轮询获取更新时请求的超时 (单位为秒)。'),
+    pollingTimeout: Schema.natural().role('time').default(25).description('通过长轮询获取更新时请求的超时 (单位为秒)。'),
```
Comment (shigma) [inline adapters/telegram/src/polling.ts:86]: I would prefer using ms as unit, changing code to something like `Math.ceil(pollingTimeout / 1000)`.

This config use `role('time')` which implies a unified time unit.

Comment (idanran) [reply adapters/telegram/src/polling.ts]: > I would prefer using ms as unit, changing code to something like `Math.ceil(pollingTimeout / 1000)`.
> 
> This config use `role('time')` which implies a unified time unit.

Changed

File: adapters/telegram/src/polling.ts [hunk 2]
```diff
@@ -83,7 +83,7 @@ export namespace HttpPolling {
 
   export const Config: Schema<Config> = Schema.object({
     protocol: Schema.const('polling' as const).required(),
-    pollingTimeout: Schema.natural().role('time').default(Time.minute).description('通过长轮询获取更新时请求的超时 (单位为秒)。'),
+    pollingTimeout: Schema.natural().role('ms').default(25000).description('通过长轮询获取更新时请求的超时 (单位为毫秒)。'),
```
Comment (MaikoTan) [inline adapters/telegram/src/polling.ts:86]:
```suggestion
    pollingTimeout: Schema.natural().role('ms').default(Time.second * 25).description('通过长轮询获取更新时请求的超时 (单位为毫秒)。'),
```

You could still be able to use a time unit here, which is much clear than counting how many zero-es.

File: adapters/telegram/src/polling.ts [hunk 3]
```diff
@@ -83,7 +83,7 @@ export namespace HttpPolling {
 
   export const Config: Schema<Config> = Schema.object({
     protocol: Schema.const('polling' as const).required(),
-    pollingTimeout: Schema.natural().role('time').default(Time.minute).description('通过长轮询获取更新时请求的超时 (单位为秒)。'),
+    pollingTimeout: Schema.natural().role('ms').default(Time.second * 25).description('通过长轮询获取更新时请求的超时 (单位为毫秒，这里使用 1000 的倍数)。'),
```
Comment (MaikoTan) [inline adapters/telegram/src/polling.ts:86]:
```suggestion
    pollingTimeout: Schema.natural().role('ms').default(Time.second * 25).description('通过长轮询获取更新时请求的超时 (单位为毫秒)。'),
```

No no no nope, you are misdirecting people.
The user SHOULD write `1000` to indicate one second, what I said above is meaning that you should keep it simple for code contributors or maintainers, not for a regular user.

Also, you should change the `pollingTimeout / Time.second` back to `pollingTimeout / 1000` as before.

**Review by MaikoTan — APPROVED** [review]

**Review by shigma — APPROVED** [review]

**Pull request merged**